libfreerdp/core/gateway/http.c

```diff
@@ -1041,3 +1097,54 @@ TRANSFER_ENCODING http_response_get_transfer_encoding(HttpResponse* response)
 
 	return response->TransferEncoding;
 }
+
+BOOL http_response_is_websocket(HttpContext* http, HttpResponse* response)
+{
+	if (!http || !response)
+		return FALSE;
+	if ((http->websocketUpgrade == TRUE) &&
```

Comment:
I'd suggest to invert the condition to do some early exits, that would make the code easier to read.
```C
if (!http->websocketUpgrade || response->StatusCode != HTTP_STATUS_SWITCH_PROTOCOLS)
   return FALSE;
```
and so on